### ISSUE TITLE
Critical & High Security vulnerability issue with Trivy Scan in postgres 16

### ISSUE THREAD

**Issue by sathyendranv** [docker-library/postgres#1297]
Hi Team, 

Postgres 16.x and 17.x has some Critical and high vulnerability reported. Can you confirm if there are any fixes planned in pipelines or these issues are false positive. 

Thanks.

Postgres 17.2:
![image](https://github.com/user-attachments/assets/8c95ea2b-d486-45ce-9f82-3d8e1266b06c)

Postgres 16.6:
![image](https://github.com/user-attachments/assets/6761a741-5bfd-4593-82c0-895312f944f9)

**Comment by tianon** [comment]
See https://github.com/docker-library/postgres/issues/1295#issuecomment-2515347464

(Duplicate of #1271 Duplicate of #1292 Duplicate of #1295)

**Comment by probitcarwyn** [comment]
While I completely agree that the _root cause_ here is the  naive "security scanners" inability to delve into the details and work out if there is a real vulnerability, the reality is that this is where many of these tools are at the moment.

One such "naive" scanner is Docker Scout. A glance at the Docker Hub tags page for this image lights up like a Christmas tree with High and Critical (all of which I believe are false positive):

- https://hub.docker.com/_/postgres/tags

As an image that is part of the [Docker Official Image](https://docs.docker.com/trusted-content/official-images/) which is supposed to [serve as the starting point for the majority of users](https://docs.docker.com/trusted-content/official-images/), is it realistic to expect users to know they need to delve into `govulncheck` to figure out if the vulnerability is real? Given this is a postgres container, they may not even be using Go in their toolchain.

The [Docker Official Image](https://docs.docker.com/trusted-content/official-images/) page goes on to say:

> The images are some of the [most secure images](https://www.docker.com/blog/enhancing-security-and-transparency-with-docker-official-images/) on Docker Hub. This is particularly important as Docker Official Images are some of the most popular on Docker Hub. Typically, Docker Official images have few or no packages containing CVEs.

So there's a certain expectation being built up here, which is at odds with what users see via the Docker Scout output.

While it certainly seems true that these reports are false positive, that's not the perception people will get from what is arguably the primary location people will be picking this image up from (Docker Hub).

Is there a way to prod Docker Scout to be smarter or otherwise mark false positives there?

**Comment by abhisheke1mishra** [comment]
I agree about false positive but my Org doesn't. They kept publishing as a Vul and asked me to fix before we put it in Prod.
So as a last resort, I took the gosu source code (Since its open source) and build it on go 1.22.4 which is free of vulnerabilities and used the updated gosu in my postgres build, which came out vulnerability free.
Only problem is I've to this 1 additional step every time I want to update to latest version.

**Comment by jesperronn** [comment]
@abhisheke1mishra 

> So as a last resort, I [built gosu] on go 1.22.4 which is free of vulnerabilities and used the updated gosu in my postgres build, which came out vulnerability free.

Thanks for your workaround. I agree that our companies vuln policies are stricter and it takes unnecessary resources if everybody have to assess false positives all the time.

If possible, could you: 

1. please provide the example code workaround here
2. please suggest a PR for this project with the updated build on go 1.22.4 or whatever is the most recent
